src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -5173,6 +5174,37 @@ def managed_db_log_replay_start(
         parameters=kwargs)
 
 
+def managed_db_log_replay_stop(
+        cmd,
+        client,
+        database_name,
+        managed_instance_name,
+        resource_group_name):
+    '''
+    Stop log replay restore.
+    '''
+
+    restore_details_client = get_sql_managed_database_restore_details_operations(cmd.cli_ctx, None)
+
+    # Determine if managed DB was created using log replay service, raise exception if not
+    restore_details = restore_details_client.get(
+        database_name=database_name,
+        managed_instance_name=managed_instance_name,
+        resource_group_name=resource_group_name,
+        restore_details_name=RestoreDetailsName.DEFAULT)
+
+    # If type is present, it must be lrsrestore in order to proceed with stop-log-replay
+    if (hasattr(restore_details, 'type_properties_type') and restore_details.type_properties_type.lower() != 'lrsrestore'):
+        raise CLIError(
+            f'Cannot stop the log replay as database {database_name} on the instance {managed_instance_name} '
+            f'in the resource group {resource_group_name} was not created with log replay service.')
```

Comment:
I do agree the design is a bit questionable because (at the moment) this command is equivalent to `az sql midb delete`. We're not calling a different API - so in that sense we can't move the logic to the backend (without making a new API). But turns out this cmd is not just a syntactic sugar for `midb delete` and should have a narrower scope - thus the change
ps. we've already made this change in powershell https://github.com/Azure/azure-powershell/pull/18409 - while sdk doesn't require any changes